crates/model/src/signature.rs

```diff
@@ -183,7 +183,9 @@ impl Signature {
                 Ok(recovered.signer)
             }
             Self::Eip1271(_) => Ok(H160::from_slice(&signature[..20])),
-            Self::PreSign => Ok(H160::from_slice(signature)),
+            // We currently don't allow solvers to pass the owner of the JIT order specifically
+            // so there is no way to know who owns a JIT pre-sign order.
+            Self::PreSign => anyhow::bail!("can't recover owner of pre-sign JIT order"),
```

Comment:
Not really related to CoW AMM orders. Just noticed this while investigating the `EIP-1271` signature issue.

Reply:
This is actually wrong and causes the unit test to fail. Will revert.